dev/com.ibm.ws.security.audit.source/src/com/ibm/ws/security/audit/source/utils/AuditUtils.java

```diff
@@ -81,62 +81,69 @@ public static String getSessionID(HttpServletRequest req) {
         final HttpServletRequest f_req = req;
 
         try {
-            sessionID = AccessController.doPrivileged(new PrivilegedExceptionAction<String>() {
-                @Override
-                public String run() throws Exception {
-                    HttpSession session = f_req.getSession();
-                    if (session != null) {
-                        return session.getId();
-                    } else {
-                        return null;
-                    }
-                }
-            });
-        } catch (PrivilegedActionException e) {
-            if ((e.getException()) instanceof com.ibm.websphere.servlet.session.UnauthorizedSessionRequestException) {
-                if (!req.isRequestedSessionIdFromCookie()) {
-                    sessionID = AccessController.doPrivileged(new PrivilegedAction<String>() {
-                        @Override
-                        public String run() {
-                            return f_req.getSession().getId();
-                        }
-                    });
-                } else {
-                    sessionID = AccessController.doPrivileged(new PrivilegedAction<String>() {
-                        @Override
-                        public String run() {
-                            return f_req.getRequestedSessionId();
-                        }
-                    });
-                }
+            sessionID = getSessionIDPrivileged(f_req);
+        } catch (PrivilegedActionException | com.ibm.websphere.servlet.session.UnauthorizedSessionRequestException e) {
+            sessionID = handleUnauthorizedSessionRequest(f_req);
+        }
+        return sessionID;
+    }
 
+    /**
+     * A helper method called from getSessionID(HttpServletRequest req) 
+     * Calls getSession() with access control handling. 
+     * 
+     * @param req
+     * @return sessionid or null 
+     * @throws PrivilegedActionException
+     */
+    private static String getSessionIDPrivileged(HttpServletRequest req) throws PrivilegedActionException {
+        return AccessController.doPrivileged((PrivilegedExceptionAction<String>) () -> {
+            HttpSession session = req.getSession(false); // Ensure the audit code does not create a new session
+            return (session != null) ? session.getId() : null;
+        });
+    }
+    /**
+     * 
+     * A helper method called from getSessionID(HttpServletRequest req)
+     * Handles exceptions 
+     * 
+     * @param req
+     * @return sessionid, null or "UnauthorizedSessionRequest"
+     */
+    private static String handleUnauthorizedSessionRequest(HttpServletRequest req) {
+        try {
+            if (!req.isRequestedSessionIdFromCookie()) {
+                return getPrivilegedSessionID(req, false); // Ensure the audit function does not create a new session
+            } else {
+                return getPrivilegedRequestedSessionID(req);
             }
```

Comment:
```suggestion
            if (!req.isRequestedSessionIdFromCookie()) {
                return "UnauthorizedSessionRequest";
            } else {
                return getPrivilegedRequestedSessionID(req);
            }
```

It makes little sense to call HttpServletRequest.getSession(false).getId() again.